backends/nxp/neutron_partitioner.py

```diff
@@ -0,0 +1,327 @@
+        return "cluster" in node.meta
+
+    def _is_node_call_function(self, node: torch.fx.node.Node):
+        return node.op == "call_function"
+
+    def is_node_delegatable(self, node: torch.fx.node.Node):
+        if self.operators_not_to_delegate != [""]:
+            any_non_delegatable = any(
+                x in node.name for x in self.operators_not_to_delegate
+            )
+            return not any_non_delegatable
+        return True
+
+    def _is_node_supported_compute(self, node: torch.fx.node.Node) -> bool:
+        """
+        Operator checking function for compute nodes.
+        """
+        if not self.is_node_delegatable(node):
+            return False
+
+        if (node_converter := supported_ops.get(node.target, None)) is None:
+            # There is no `NodeConverter` for this `node`.
+            return False
+
+        return (
+            self._is_node_call_function(node)
+            and self._is_node_quantized(node)
+            and
+            # TODO: `view_copy` node should be delegated only if it's not the only operator in the cluster.
+            node_converter.is_supported(node, self.target, self.parameters_mapping)
+        )
+
+    def _is_node_supported_non_compute(self, node: torch.fx.node.Node) -> bool:
+        """
+        If the node is a quantize, dequantize or auxiliary node inside a QDQ cluster, the support on Neutron
+        is determined by the support of the compute operator.
+        """
+        return self._is_node_quantized(node) and self._is_node_supported_compute(
+            self.qdq_clusters[node.meta["cluster"]].compute_node
+        )
+
+    def is_node_supported(
+        self, submodules: Mapping[str, torch.nn.Module], node: torch.fx.Node
+    ) -> bool:
+        """
+        Check if the Edge operator is supported on Neutron.
+        """
+
+        if (
+            QDQClusterRecognizer.is_quant_node(node)
+            or QDQClusterRecognizer.is_dequant_node(node)
+            or QDQClusterRecognizer.is_auxiliary_node(node)
+        ):
+            return self._is_node_supported_non_compute(node)
+        else:
+            return self._is_node_supported_compute(node)
+
+
+@final
+class NeutronPartitioner(Partitioner):
+    def __init__(self, compile_spec: List[CompileSpec]) -> None:
+        self.delegation_spec = DelegationSpec(NeutronBackend.__name__, compile_spec)
+
+    def partition(self, exported_program: ExportedProgram) -> PartitionResult:
+        # Run the CapabilityBasedPartitioner to return the largest possible
+        # subgraphs containing the nodes with the tags
+        logging.info("NeutronPartitioner::partition")
+        partition_tags = {}
+
+        graph_module = exported_program.graph_module
+        nodes = list(graph_module.graph.nodes)
+
+        qdq_clusterer = QDQClusterRecognizer()
+
+        qdq_clusterer.tag_qdq_clusters(nodes)
+
+        graph_module.recompile()
+        target = self.delegation_spec[1][2].value
```

Comment:
can we use `CompileSpec.key : str` to make this more readable?

Reply:
:white_check_mark: